python/taulabs/telemetry.py

```diff
@@ -307,14 +361,49 @@ def __init__(self, host="127.0.0.1", port=9000, *args, **kwargs):
 
         FDTelemetry.__init__(self, fd=s.fileno(), *args, **kwargs)
 
+# TODO XXX : Plumb appropriate cleanup / file close for these classes
+
+class SerialTelemetry(FDTelemetry):
+    """ Serial telemetry interface """
+    def __init__(self, port, speed=None, *args, **kwargs):
+        """ Creates telemetry instance talking over (real or virtual) serial port.
+        
+         - port: Serial port path
+         - speed: Baud rate (doesn't really matter for VCP, defaults 115200)
+
+        Meaningful parameters passed up to TelemetryBase include: githash,
+        service_in_iter, iter_blocks, use_walltime
+        
+        Requires the module pyserial to provide OS independance.
+        """
+
+        import serial
+
+        if speed is None:
+            speed=115200
+
+        ser = serial.Serial(port, speed)
+
+        FDTelemetry.__init__(self, fd=ser.fileno(), *args, **kwargs)
+
 class FileTelemetry(TelemetryBase):
-    # TODO accept file object
-    def __init__(self, filename='sim_log.tll', parseHeader=False,
+    """ Telemetry interface to data in a file """
+
+    def __init__(self, file_obj, parse_header=False,
```

Comment:
I think it would be good to default to `use_walltime=False` for file based telemetry. I was using the new Python module  (great job @mlyle !) and was really confused by the strange time stamps. I.e., if there are timestamps in the file, it should use them by default.